com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/RecentLogsRetriever.java

```diff
@@ -24,7 +24,12 @@
 
     public List<ApplicationLog> getRecentLogs(CloudControllerClient client, String appName, LogsOffset offset) {
         List<ApplicationLog> appLogs = client.getRecentLogs(appName);
-        return offset == null ? appLogs : ListUtils.select(appLogs, appLog -> isLogNew(appLog, offset));
+        if (offset == null) {
+            return appLogs;
+        }
+        return appLogs.stream()

```

Comment:
Back when I first refactored it to use `ListUtils.select` I didn't know the performance and scalability difference between it and using the stream api. Now I understand that the `select` method scales linearly while streams scale better the larger the input data. And because application logs are usually quite a few, I decided to revert it to the way it was.